### PR TITLE
build: extract interface and util sub compilation from core

### PR DESCRIPTION
**Pull request by mhevery**
Broke down the monolithic `@angular/core` into three smaller `ts_library` rules
- `//packages/core/src/interface`
- `//packages/core/src/di/interface`
- `//packages/core/src/util`

This is initial work to do further breakdown of the compilation.

### CODE REVIEW

File: packages/core/src/di/interface/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (IgorMinar) [inline packages/core/src/di/interface/BUILD.bazel]: this doesn't seem right. we don't want the visibility to be public. use the following instead:
```suggestion
package(default_visibility = ["//packages/core:__subpackages__"])
```

Please refactor all the internal build files to use this restricted visibility.

Comment (mhevery) [reply packages/core/src/di/interface/BUILD.bazel]: fixed

File: packages/core/src/change_detection/change_detection_util.ts [hunk 2]
```diff
@@ -6,7 +6,8 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {getSymbolIterator, looseIdentical} from '../util';
+import {looseIdentical} from '../util/comparison';
+import {getSymbolIterator} from '../util/symbol';
```
[the remark this answers is not in the document]
Comment (IgorMinar) [reply packages/core/src/change_detection/change_detection_util.ts]: after reviewing the rest of the PR I now believe that this was intentional and it's the right thing to do given the variety of util helpers in this directory. thanks for being patient with me :)

**Review by IgorMinar — APPROVED** [review]
one suggestion, the rest looks good.

**Pull request closed without merging**